tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorValidationTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Linq;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Translators
+{
+    public class PredicateTranslatorValidationTests
+    {
+        private readonly IMongoCollection<TestObject> _collection;
+
+        public PredicateTranslatorValidationTests()
+        {
+            var connectionString = CoreTestConfiguration.ConnectionString.ToString();
+            IMongoClient client = new MongoClient(connectionString);
+            var database = client.GetDatabase("test");
+            _collection = database.GetCollection<TestObject>("testObject");
+            database.DropCollection("testObject");
```

Comment:
I frown on test setup code in a constructor because when looking at a test it's just too easy to not realize that there is some setup code being run before the test method itself runs.

My preferred solution is to have a Setup() method (or named differently if more appropriate) and call it from each test method.

This has the advantage of making each test self sufficient (no dependency on constructor code). It also means you can have more than one setup method if different tests require different setups (or omit the call to Setup if a particular test requires no setup).

Reply:
It's typical behavior for unit tests to have some test preparation steps. Usually, it happens by [Setup] or similar attribute which works out of the box. But xUnit framework has a different position about this question and, as i see, they recommend to use a c# constructor for this purpose: https://xunit.github.io/docs/comparisons.
However, I understand your point of view the same as author's of this post https://jamesnewkirk.typepad.com/posts/2007/09/why-you-should-.html.
I will add `Setup` method.